Refactor(input model): create the input

grysgreat opened the pull request:
rebuild class and function. This has no effect on the current code.
 Refactor #2549

grysgreat committed:
Refactor(input model): create the input
 rebuild class and function. This has no effect on the current code.
 Refactor #2549

grysgreat commented:
Refact the input model. - Gu Hongyaoxing

mohanchen commented on source/module_io/parameter_pool.h:
Use English notes.

mohanchen commented on source/module_io/parameter_pool.h:
```diff
@@ -0,0 +1,116 @@
+        /**
+         * @brief ���캯���������ò������ͺ�Ĭ�ϲ���ֵ
+         * 
+         * @param t ��������ö��ֵ
```
Use English for all of the notes in ABACUS

grysgreat committed:
Refactor(input model):
create the input & using En note
 rebuild class and function. This has no effect on the current code.
 Refactor #2549

grysgreat committed:
Refactor(input model): create the input
 rebuild class and function. This has no effect on the current code.
 Refactor #2549

hongriTianqi commented on source/module_io/parameter_pool.cpp:
```diff
@@ -0,0 +1,1432 @@
+ * @brief This function is used to read the input parameter file and store it as a key-value pair
+ * @param fn Enter the path to the parameter file
+ */
+bool input_parameters_get(const std::string& fn)
```
Please add unit tests for all the newly added functions.

hongriTianqi commented on source/module_io/parameter_pool.cpp:
```diff
@@ -0,0 +1,1432 @@
+ * @param sa Pointer to the string to be converted
+ * @param sb Store a pointer to the converted string, which needs to be pre-allocated with enough space
+ */
+void strtolower(char* sa, char* sb)
```
Please write the new functions inside the namespace ModuleIO

hongriTianqi reviewed (changes requested):
Please remove the empty git file

grysgreat committed:
Merge branch 'deepmodeling:develop' into abacus-input
